src/PointwiseFunctions/Xcts/LongitudinalOperator.cpp

```diff
@@ -0,0 +1,89 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/Elasticity/PotentialEnergy.hpp"
+
+#include <algorithm>
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Xcts {
+
+/// \cond
+template <typename DataType>
+void longitudinal_operator(const gsl::not_null<tnsr::II<DataType, 3>*> result,
+                           const tnsr::ii<DataType, 3>& strain,
+                           const tnsr::II<DataType, 3>& inv_metric) noexcept {
+  for (size_t i = 0; i < 3; ++i) {
+    for (size_t j = 0; j <= i; ++j) {
+      // Unroll first iteration of the loop over `k` to avoid filling the result
+      // with zero initially
```

Comment:
```suggestion
// k=l=0  of inner loop as assignment to avoid filling the result with zero initially
```

(I was confused by the orig comment, as not only k was unrolled, but also l)